### PR TITLE
Issue 368 udpgen dream

### PR DESCRIPTION
**Pull request by mortenjc**
### Issue reference / description

The branch you merge from should already reference an event-formation-unit github ticket number. You can add a descriptive title, but if an issue is referenced, you don't have to.

## Checklist for submitter

- [ ] Check for conflict with integration test
- [ ] Unit tests pass

---

## Nominate for Group Code Review (Anyone can nominate it)
Indicate if you think the code should be reviewed in a Thursday code review session.

- [ ] Recommend for group code review

Also, nominate it on the code_review Slack channel.

### CODE REVIEW

**Comment by mortenjc** [comment]
Mostly I havce added a new EFU module for dream - the old jalousie pipeline used different data and mappings.
So new pipeline, unit tests, cmake magic, udp generator for simulated dream data.

File: src/modules/dream/test/DreamBaseTest.cpp [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+/// | Data header 1 | Readout 1 | Readout 2 | Readout 3 |
+/// | Data header 2 |
+/// | Data block 1  |
+/// | Data header 1 |
+/// | Data block 1  |
+///
+std::vector<uint8_t> TestPacket2{
+    // ESS header
+                0x00, 0x00, // pad, v0
+    0x45, 0x53, 0x53, 0x60, //  'E' 'S' 'S' 0x00
```
Comment (amues) [inline src/modules/dream/test/DreamBaseTest.cpp:55]: Does the comment say the last byte should be 0x00?

**Review by amues — APPROVED** [review]
I've left some comments, feel free to merge after addressing or ignoring them if they don't make sense.

File: src/modules/dream/DreamBase.h [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+
+
+
+
+
+  /// \todo figure out the right size  of EthernetBufferMaxEntries
+  static const int EthernetBufferMaxEntries {2000};
+  static const int EthernetBufferSize {9000}; /// bytes
+  static const int KafkaBufferSize {124000}; /// entries ~ 1MB
```
[the remark this answers is not in the document]
Comment (mortenjc) [reply src/modules/dream/DreamBase.h:44]: Yes it is - there is a lot of other refactoring that needs to be done - eventually. But I'll leve it fornow

File: src/modules/dream/test/DreamBaseTest.cpp (hunk 1 above)
Comment (mortenjc) [reply src/modules/dream/test/DreamBaseTest.cpp:55]: well spotted, will fix

File: src/modules/dream/test/DataParserTestData.h [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00,
+    0x01, 0x03, 0x02, 0x03,
+    0x03, 0x03, 0x04, 0x03,
+};
+
+std::vector<uint8_t> OkThreeDreamReadouts
+{
+  // Data Header 1
+  0x00, 0x00, 0x28, 0x00, // ring 0, fen 1, data size 40 bytes
```
[the remark this answers is not in the document]
Comment (mortenjc) [reply src/modules/dream/test/DataParserTestData.h:51]: well spotted, will fix

**Pull request merged**